src/Listing/Cell/DataCell.tsx

```diff
@@ -0,0 +1,193 @@
+
+import { equals } from 'ramda';
+
+import { makeStyles, Tooltip, Typography, Theme } from '@material-ui/core';
+
+import {
+  Column,
+  ColumnType,
+  ComponentColumnProps,
+  RowColorCondition,
+} from '../models';
+
+import Cell from '.';
+
+interface Props {
+  row?;
+  isRowSelected: boolean;
+  isRowHovered: boolean;
+  rowColorConditions?: Array<RowColorCondition>;
+  listingCheckable: boolean;
+  column: Column;
+}
+
+const useStyles = makeStyles<Theme, { listingCheckable: boolean }>(() => ({
+  cell: {
+    alignSelf: 'stretch',
+    display: 'flex',
+    alignItems: 'center',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+  },
+  text: {
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+}));
+
+const DataCell = ({
+  row,
+  column,
+  listingCheckable,
+  isRowSelected,
+  isRowHovered,
+  rowColorConditions,
+}: Props): JSX.Element | null => {
+  const classes = useStyles({ listingCheckable });
+
+  const commonCellProps = {
+    isRowHovered,
+    rowColorConditions,
+    className: classes.cell,
+    align: 'left' as const,
+    row,
+  };
+
+  const cellByColumnType = {
+    [ColumnType.string]: (): JSX.Element => {
+      const { getFormattedString, isTruncated, getColSpan } = column;
+
+      const colSpan = getColSpan?.(isRowSelected);
+
+      const formattedString = getFormattedString?.(row) || '';
+
+      const gridColumn = colSpan ? `auto / span ${colSpan}` : 'auto / auto';
+
+      const typography = (
+        <Typography variant="body2" className={classes.text}>
+          {formattedString}
+        </Typography>
+      );
+
+      return (
+        <Cell style={{ gridColumn }} {...commonCellProps}>
+          {isTruncated && (
+            <Tooltip title={formattedString}>{typography}</Tooltip>
+          )}
+          {!isTruncated && typography}
+        </Cell>
+      );
+    },
+    [ColumnType.component]: (): JSX.Element | null => {
+      const { getHiddenCondition, clickable } = column;
+      const Component = column.Component as (
+        props: ComponentColumnProps,
+      ) => JSX.Element;
+
+      const isCellHidden = getHiddenCondition?.(isRowSelected);
+
+      if (isCellHidden) {
+        return null;
+      }
+
+      return (
+        <Cell
+          onClick={(e): void => {
+            if (!clickable) {
+              return;
+            }
+            e.preventDefault();
+            e.stopPropagation();
+          }}
+          {...commonCellProps}
+        >
+          <Component
+            row={row}
+            isSelected={isRowSelected}
+            isHovered={isRowHovered}
+          />
+        </Cell>
+      );
+    },
+  };
+
+  return cellByColumnType[column.type]();
+};
+
+const MemoizedDataCell = React.memo<Props>(DataCell, (prevProps, nextProps) => {
+  const previousColumn = prevProps.column;
+  const previousRow = prevProps.row;
+  const previousIsRowHovered = prevProps.isRowHovered;
+  const previousIsRowSelected = prevProps.isRowSelected;
```

Comment:
we don't have rule for this now, but I think it's more readable like this : 
```suggestion
  const {
    column: previousColumn,
    row: previousRow,
    isRowHovered: previousIsRowHovered,
    isRowSelected: previousIsRowSelected,
  } = prevProps;
```